File: ObjCSDKVerification/ObjCSDKVerification/TableViewVC.m
//
//  TableViewVC.m
//  ObjCSDKVerification
//
//  Created by Siarhei Barmotska on 20/02/2017.
//  All rights reserved.
//

#import "TableViewVC.h"
#import "TableViewCell.h"
#import <AmpiriSDK/Ampiri.h>

@interface TableViewVC ()

@property (strong) AMPTableViewStreamAdapter *adapter;
@end

@implementation TableViewVC

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.adapter = [[AmpiriSDK sharedSDK]
                    createStreamAdapterForTableView: self.tableView
                    parentViewController: self
                    adUnitId: @"7f900c7d-7ce3-4190-8e93-310053e70ca2"
                    templateType: AMPNativeAdsTemplateTypeList
                    templateCustomization: ^(AMPTemplateCustomizationObject
                                             *templateCustomizationObject) {
                        templateCustomizationObject.ampCTABorderWidth = 1;
                        templateCustomizationObject.ampCTACornerRadius = 5;
                    }];
}


#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return 20;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    TableViewCell *cell = [tableView amp_dequeueReusableCellWithIdentifier:NSStringFromClass([TableViewCell class]) forIndexPath:indexPath];
    
    cell.nameLabel.text = [NSString stringWithFormat:@"#%ld", indexPath.row];
    return cell;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    return 44;
}

@end

File: SwiftSDKVerification/SwiftSDKVerification/TableViewVC.swift
//
//  TableViewVC.swift
//  SwiftSDKVerification
//
//  Created by Siarhei Barmotska on 21/02/2017.
//  All rights reserved.
//

import UIKit
import AmpiriSDK

class TableViewVC: UITableViewController {
    fileprivate var adapter: AMPTableViewStreamAdapter?

    override func viewDidLoad() {
        super.viewDidLoad()
        self.adapter = AmpiriSDK.shared().addLocationControl(to: tableView,
                                                             parentViewController: self,
                                                             adUnitId: "7f900c7d-7ce3-4190-8e93-310053e70ca2",
                                                             templateType: .inFeed,
                                                             templateCustomization: { templateCustomizationObject in
                                                                templateCustomizationObject.ampCTABorderWidth = 1
                                                                templateCustomizationObject.ampCTACornerRadius = 5
                                                            })
    }


    // MARK: - Table view data source

    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 20
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.amp_dequeueReusableCell(withIdentifier: "MyTableViewCell", for: indexPath) as! MyTableViewCell
        cell.nameLabel.text = "#\(indexPath.row)"
        return cell
    }
    
    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 44
    }
 
}
